web/src/reducers/budget.js

```diff
@@ -250,113 +254,172 @@ const buildBudget = bigState => {
       });
     });
 
-    /**
-     * Get the amount of "other funding" attributed to each type of
-     * cost for a given year of the current activity.
-     * @description For each type of cost, this method calculates that type's
-     *    contribution to the total cost for the year.  It then applies that
-     *    same percentage to the total "other funding".  For example:
-     *
-     *    other funding is $200
-     *    the total cost is $1,000
-     *    contractor costs are $500
-     *    contractor costs is 50% of total costs
-     *    so other funding for contractor costs is 50% * $200 = $100
-     *
-     *    This function is placed here in the code because it depends on the
-     *    results of the previous calculations.  This way if anyone accidentally
-     *    uses it earlier in the code, it will be an error because the function
-     *    hasn't been defined yet.
-     * @param {String} year As a four-character year string (e.g., '2018')
-     * @returns {{contractors:Number, expenses:Number, statePersonnel:Number}} Other funding for each type of cost
-     */
-    const getOtherFunding = year => {
-      const otherTotal = n(allocation[year].other);
-
-      const costs = newState[fundingSource];
-      const yearTotal = costs.combined[year].total;
-
-      return {
-        contractors: otherTotal * costs.contractors[year].total / yearTotal,
-        expenses: otherTotal * costs.expenses[year].total / yearTotal,
-        statePersonnel:
-          otherTotal * costs.statePersonnel[year].total / yearTotal
-      };
-    };
+    // Converts an array where the values correspond to contractors,
+    // expenses, and state personnel values, respectively.
+    const costShareReducer = (acc, value, i) => ({
+      ...acc,
+      [['contractors', 'expenses', 'statePersonnel'][i]]: value
+    });
 
     // Now loop back over the years and compute state and federal shares
     // of all the costs.
     years.forEach(year => {
-      const other = getOtherFunding(year);
+      const totalCost = activityTotals.data.combined[year];
+      const totalGovShare = totalCost - n(allocation[year].other);
+      const costShares = getAllocation(year, totalGovShare);
```

Comment:
100% open to better names for these variables.  `totalCostShare` should maybe be `totalMedicaidShare`, since it represents the total amount that Medicaid is responsible for, whether at the state or federal level.

`costShares` is the Medicaid total share, broken into state and federal shares.

Reply:
Oops, yes, the latter.  I'll go make these changes!